MFVNeutralino/plugins/VertexHistos.cc

```diff
@@ -382,10 +428,131 @@ void MFVVertexHistos::analyze(const edm::Event& event, const edm::EventSetup&) {
   const int nsv = int(auxes->size());
   h_nsv->Fill(nsv, w);
 
+  for (int igenv = 0; igenv < 2; ++igenv) {
+    double genx = mevent->gen_lsp_decay[igenv*3+0];
+    double geny = mevent->gen_lsp_decay[igenv*3+1];
+    double genz = mevent->gen_lsp_decay[igenv*3+2];
+    double genbs2ddist = mevent->mag(genx - mevent->bsx_at_z(genz),
+                                     geny - mevent->bsy_at_z(genz) 
+        );
+    h_genbs2ddist->Fill(genbs2ddist, w);
+  }
```

Comment:
Would this be better as an event-level histogram? I don't actually know--I guess if we e.g. have an event with a single 4-trk vertex and we look at that corresponding set of VertexHistos, it can be interesting to see the gen-level displacements look like, and could make the comparisons across different track and vertex multiplicities. Maybe it gets too muddled with there being two LLPs per event though, not sure.

Reply:
Sorry, what does "event-level histogram" mean? I am not very clear about what modification I should do...

Reply:
Sorry, I was thinking this might make more sense in EventHistos (since it pertains to the events) rather than VertexHistos (which looks mostly at vertex-level quantities). I think as it stands, it might fill the plot once for each SV, which may be less interesting than seeing it once per event?

Reply:
I will try to move that to EventHistos. Thanks!